contrib/guix/libexec/build.sh

```diff
@@ -71,16 +71,12 @@ unset CPLUS_INCLUDE_PATH
 unset OBJC_INCLUDE_PATH
 unset OBJCPLUS_INCLUDE_PATH
 
-export LIBRARY_PATH="${NATIVE_GCC}/lib:${NATIVE_GCC}/lib64:${NATIVE_GCC_STATIC}/lib:${NATIVE_GCC_STATIC}/lib64"
+export LIBRARY_PATH="${NATIVE_GCC}/lib:${NATIVE_GCC_STATIC}/lib"
```

Comment:
possible to do only after bitcoin#24031

> 1. The `prepend_to_search_env_var()` function was introduced in https://github.com/bitcoin/bitcoin/commit/c1ae726a13ecfa5e7e9fdc3030a8110b8bb263f8, and it has no longer been used since https://github.com/bitcoin/bitcoin/commit/1dd8cbfbc6318308b5e75023568f818ef6a9c7e4.